signac/core/rediscollection.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Implements Redis-backend.
+
+This implements the Redis-backend for SyncedCollection API by
+implementing sync and load methods.
+"""
+import json
+
+from .synced_collection import SyncedCollection
+from .syncedattrdict import SyncedAttrDict
+from .synced_list import SyncedList
+
+
+class RedisCollection(SyncedCollection):
+    """Implement sync and load using a Redis backend."""
+
+    backend = __name__  # type: ignore
+
+    def __init__(self, name=None, client=None, redis_kwargs=None, **kwargs):
+        import redis
+
+        if client is None:
+            redis_kwargs = redis_kwargs if redis_kwargs is not None else {}
+            self._client = redis.Redis(**redis_kwargs)
+        else:
+            self._client = client
+        self._name = name
+        super().__init__(**kwargs)
+        if (name is None) == (self._parent is None):
+            raise ValueError(
+                "Illegal argument combination, one of the two arguments, "
+                "parent or name must be None, but not both.")
+
+    def _load(self):
+        """Load the data from a Redis-database."""
+        blob = self._client.get(self._name)
+        return json.loads(blob) if blob is not None else blob
```

Comment:
```suggestion
        return None if blob is None else json.loads(blob)
```
Slightly easier to interpret I think.